ably/realtime/realtime_channel.py

```diff
@@ -97,6 +102,56 @@ async def detach(self):
         await self.__detach_future
         self.set_state(ChannelState.DETACHED)
 
+    async def subscribe(self, *args):
```

Comment:
The listener reference should be retained until it's `unsubscribe'd, or if the reference to the channel is lost (ie if the user clears up their own references and calls `Channels.release`)

As for the illustration, we're planning on addressing that very soon. I think it would be better to take a holistic approach and think carefully about how we present this stuff so I'd prefer not to start doing it in this PR. I would say that at least for reviewers the test fixtures should be enough to see how it's used and what it's doing

If it helps, calling `subscribe` here with a coroutine function is essentially the same as using an async function as a subscribe listener in ably-js

Reply:
A holistic approach is understandable, however bear in mind that this does make it less easy for reviewers of this pull request to envisage the usage.

Also, could you perhaps add an issue to the backlog for the milestone you're working on to make sure that examples don't get forgotten before we do our first release?

Reply:
We do have a Jira item for it (SDK-2656) but it wasn't created in GitHub for some reason, but don't worry it's in the current sprint so we def won't forget :)